packages/forms/backend/sirius-components-widget-reference/src/main/java/org/eclipse/sirius/components/widget/reference/ReferenceWidgetComponent.java

```diff
@@ -87,10 +188,20 @@ public Element render() {
                         };
                         referenceValueBuilder.clickHandler(clickHandler);
                     }
+                    if (removeHandlerProvider != null) {
+                        Supplier<IStatus> removeHandler = () -> {
+                            return removeHandlerProvider.apply(childVariables);
+                        };
+                        referenceValueBuilder.removeHandler(removeHandler);
+                    }
                     return referenceValueBuilder.build();
                 })
                 .toList();
+        return items;
+    }
 
+    private List<ReferenceValue> getOptions(VariableManager variableManager, ReferenceWidgetDescription referenceDescription) {
+        List<?> rawOptions = referenceDescription.getOptionsProvider().apply(variableManager);
         List<ReferenceValue> options = rawOptions.stream()
```

Comment:
You could directly return here, options variable is redundant

Reply:
Done